### PR TITLE
systemd: 255.6 -> 256

### PR DESCRIPTION
**Pull request by ElvishJerricco**
## Description of changes

I'll make this target staging before marking it ready for review. For now it's convenient to have it based on nixos-unstable-small.

There's a couple of patches that needed slightly more work than just a rebase, namely 0012, 0014, and 0018 (to an even lesser degree). Other than that, a few more things became dlopen dependencies, and now `systemd-dissect` has some funky new option requiring `libarchive`.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ElvishJerricco** [comment]
It seems like we're not getting the new `.note.dlopen` ELF NOTEs. Not sure what's up with that.

**Comment by ElvishJerricco** [comment]
nvm, this was already done in #307068

**Comment by nyabinary** [comment]
> nvm, this was already done in #307068

Tbf that has a merge conflict though :P

**Comment by ElvishJerricco** [comment]
yea, but improvements can be made on the original PR instead of having a duplicate one. If @nikstur wants to combine our branches, that's fine with me :P

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -263,6 +265,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   postPatch = ''
     substituteInPlace src/basic/path-util.h --replace "@defaultPathNormal@" "${placeholder "out"}/bin/"
+    substituteInPlace src/basic/path-util.c --replace "@defaultPathNormal@" "${placeholder "out"}/bin/"
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/systemd/default.nix:268]: We should replace those with --replace-fail

**Pull request closed without merging**